### PR TITLE
[DropbearServerPlugin] Initial version for review

### PR DESCRIPTION
**Pull request by s-rapolu**
DropbearServer plugin implementation related changes in WPEFrameworkPlugins (ThunderNanoServices) repository

### CODE REVIEW

File: DropbearServer/DropbearServer.cpp [hunk 1]
```diff
@@ -17,42 +17,76 @@ namespace Plugin {
...
+    // GET ../Window <- Return the Rectangle in which the player is running.
+    // POST ../Create/<Type> <- Create an instance of a stream of type <Type>, Body return the stream index for reference in the other calls.
+    // PUT ../<Number>/Load <- Load the URL given in the body onto this stream
+    // PUT ../<Number>/Attach <- Attach a decoder to the primer of stream <Number>
+    // PUT ../<Number>/Speed/<speed> <- Set the stream as a speed of <speed>
+    // PUT ../<Number>/Detach <- Detach a decoder to the primer of stream <Number>
+    // PUT ../<Number>/Window/X/Y/Width/Height <- Set Window size
+    // DELETE ../<Number> <- Drop the stream adn thus the decoder
+    //
+    // POST/PUT	<- StartService
```
Comment (pwielders) [inline DropbearServer/DropbearServer.cpp]: Do not expose starting and stoppoing of a service through the RESTFull interface. This is already part of the Thunder framework. Activating and Deactivating a plugin.

File: DropbearServer/DropbearServer.cpp [hunk 2]
```diff
@@ -17,42 +17,76 @@ namespace Plugin {
...
+    // POST/PUT	<- StopService
```
Comment (pwielders) [inline DropbearServer/DropbearServer.cpp]: See remark above. This is Deactivating the plugin.

**Comment by pwielders** [comment]
Implementation looks good, with the exception of the RESTFul/JSONRPC starting and stopping of the service. This really needs to be removed.
Furthermore, I think it might be interesting Food for Thought if this should not be called the SecureShellServer plugin with multiple (for now only 1) implementations of a SSH server implementation. Here is the DropBear implementation but what about a tinyssh.org iimplementation ?
Should fit behind a the suggested interface and might even make the system smaller and less resource consuming. Even think that the server could be run in-process, to even more reduce the resource usage, but that is just a configuration option :-)

File: DropbearServer/DropbearServer.cpp (hunk 1 above)
Comment (s-rapolu) [reply DropbearServer/DropbearServer.cpp]: Agreed. I'll make the necessary corrections.

File: DropbearServer/DropbearServer.cpp (hunk 2 above)
Comment (s-rapolu) [reply DropbearServer/DropbearServer.cpp]: Agreed

**Pull request closed without merging**